libpromises/conversion.c

```diff
@@ -409,7 +409,7 @@ long IntFromString(const char *s)
 
     remainder[0] = '\0';
 
-    sscanf(s, "%lld%c%s", &a, &c, remainder);
+    sscanf(s, "%lld%c%4095s", &a, &c, remainder);
```

Comment:
I don't see what's the role of `remainder` being a string in this function, dump it completely. If you want to check if anything follows, use another `%c` and read one more character. Also remember to check the `sscanf()` return value. This is the way to kill the potential security issue here.